src/lib/utils/plugin_manager.hpp

```diff
@@ -23,7 +25,7 @@ class PluginManager : public Singleton<PluginManager> {
   friend class SingletonTest;
 
  public:
-  void load_plugin(const std::filesystem::path& path);
+  void load_plugin(const std::filesystem::path& path, std::shared_ptr<Listenable> listenable = nullptr);
```

Comment:
I don’t like this part. What if the plugin should attach to two things? No, the plugin needs to find whatever it wants to attach to on its own.

Reply:
> The other option I though about was to make the benchmark runner a Singleton and the add_listener functions static.

Maybe #1677 would help here?

Reply:
Yes, I agree that this is not ideal. If the plugin should find what it wants to attach to on its own, I don't think we can get around making the benchmark runner a Singleton, which I'm okay with. Works with #1677 I'd say, although it doesn't really make a difference I think. We'd just have to decide that the benchmark runner is a single global instance then, either as Singleton or accessible through the Hyrise Singleton.

Reply:
I'm having some trouble converting `BenchmarkRunner` into a Singleton. The problem is the `reset()` method, which assigns a newly created `BenchmarkRunner` to the static instance. `BenchmarkRunner` however has no copy-assignment constructor. They get implicitly deleted because of some of the fields, like the `const BenchmarkConfig`, or the `unique_ptr's`. It might be a bigger change than I thought at first.